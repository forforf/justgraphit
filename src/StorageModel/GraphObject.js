
class GraphObject {

  static fromPlainObject(obj) {
    const ks = Object.keys(obj);
    if (ks.length > 1) {
      throw new Error(`Only single keyed object supported, but got ${ks}`)
    }
    const name = Object.keys(obj)[0];
    const data = obj[name];
    return new GraphObject(name, data);
  }

  constructor(graphName, graphData) {
    this.name = graphName;
    this.data = graphData;
  }
}

export default GraphObject