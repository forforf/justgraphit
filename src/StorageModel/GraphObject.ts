import {GraphName, JustGraphitEntry, JustGraphitStore} from "../JustGraphitTypes";


// Used to enforce that there is a single key/value pair (key = name, value = data in this case)
class GraphObject {
  name: GraphName;
  data: JustGraphitEntry[];

  // input object should look like a store, but actually only have a single graph -> data entry
  static fromPlainObject(obj: JustGraphitStore): GraphObject {
    const ks = Object.keys(obj);
    if (ks.length > 1) {
      throw new Error(`Only single keyed object supported, but got ${ks}`)
    }
    const name = Object.keys(obj)[0];
    const data = obj[name];
    return new GraphObject(name, data);
  }

  constructor(graphName: GraphName, graphData: JustGraphitEntry[]) {
    this.name = graphName;
    this.data = graphData;
  }
}

export default GraphObject;