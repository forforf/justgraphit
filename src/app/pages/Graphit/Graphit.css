
.Graphit {
  text-align: center;
  position:relative;
  min-width:50%;
  max-width:100%;
}

.left {
  position:relative;
  width:50%;
  float:left;
}

.right {
  position:relative;
  width: 50%;
  min-width:50%;
  max-width:100%;
  float:left;
}

.GraphChart {
  margin: 1em;
}
