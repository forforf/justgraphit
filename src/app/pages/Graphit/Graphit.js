
import React, { Component } from 'react';
import './Graphit.css';
import GraphInput from 'components/GraphInput/GraphInput'
import GraphChart from 'components/GraphViewer/GraphViewer.js'


class Graphit extends Component {
  render() {
    return (
      <div className="App">
        <GraphInput />
        <GraphChart />
      </div>
    );
  }
}

export default Graphit;
