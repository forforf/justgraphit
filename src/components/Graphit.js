import React from 'react';
import './Graphit/Graphit.css';
import GraphInput from './Graph/GraphInput'
import GraphChart from './Graph/GraphChart'
import GraphList from './Graph/GraphList'


function Graphit({graphObject, addNewNumber, changeSelectedGraph, deleteGraph, graphNameList}) {

  // props
  const graphChartProps = { graphObject }
  const graphInputProps = {
    graphName: graphObject.name,
    addNewNumber
  }
  const graphListProps = {
    graphNameList,
    changeSelectedGraph,
    deleteGraph
  }

  return (
    <div className="Graphit">
      <div className="wrapper">
        <div className="row">
          <div className="left col-2-3">
            <GraphInput {...graphInputProps}/>
            <GraphChart {...graphChartProps}/>
          </div>
          <div className="right col-1-3">
            <GraphList {...graphListProps}/>
          </div>
        </div>
      </div>
    </div>
  );
}

export default Graphit;
