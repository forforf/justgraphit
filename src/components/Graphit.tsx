import React from 'react';
import './Graphit/Graphit.css';
import GraphInput from './Graph/GraphInput';
import GraphChart from './Graph/GraphChart';
import GraphList from './Graph/GraphList';
import {
  GraphEntryHandler,
  GraphHandler,
  GraphName,
} from '../JustGraphitTypes';
import GraphObject from '../StorageModel/GraphObject';

export type GraphitProps = {
  graphObject: GraphObject; // actually a single key/value pair of the store
  addNewNumber: GraphEntryHandler;
  changeSelectedGraph: GraphHandler;
  deleteGraph: GraphHandler;
  graphNameList: GraphName[];
};

function Graphit({
  graphObject,
  addNewNumber,
  changeSelectedGraph,
  deleteGraph,
  graphNameList,
}: GraphitProps): JSX.Element {
  // props
  const graphChartProps = { graphObject };
  const graphInputProps = {
    graphName: graphObject.name,
    addNewNumber,
  };
  const graphListProps = {
    graphNameList,
    changeSelectedGraph,
    deleteGraph,
  };

  return (
    <div className="Graphit">
      <div className="wrapper">
        <div className="row">
          <div className="left col-2-3">
            <GraphInput {...graphInputProps} />
            <GraphChart {...graphChartProps} />
          </div>
          <div className="right col-1-3">
            <GraphList {...graphListProps} />
          </div>
        </div>
      </div>
    </div>
  );
}

export default Graphit;
