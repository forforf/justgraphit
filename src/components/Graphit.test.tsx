import React from 'react';
import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';
import Graphit, {GraphitProps} from './Graphit';
import GraphObject from "../StorageModel/GraphObject";

let graphitProps: GraphitProps;
beforeEach(() => {
  graphitProps = {
    graphObject: new GraphObject('empty graph', []),
    addNewNumber: jest.fn(),
    changeSelectedGraph: jest.fn(),
    deleteGraph: jest.fn(),
    graphNameList: []
  }
})

test('deep snapshot', () => {
  const component = renderer.create(
    <Graphit {...graphitProps}/>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('shallow snapshot', () => {
  const renderer = ShallowRenderer.createRenderer();
  renderer.render(<Graphit {...graphitProps}/>);
  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot()
});
