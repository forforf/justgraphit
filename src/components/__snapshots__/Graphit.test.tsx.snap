// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deep snapshot 1`] = `
<div
  className="Graphit"
>
  <div
    className="wrapper"
  >
    <div
      className="row"
    >
      <div
        className="left col-2-3"
      >
        <div
          className="GraphInput"
        >
          <div
            className="GraphInput-name"
          >
            <h3>
              empty graph
            </h3>
          </div>
          <form>
            <div
              className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
              onKeyPress={[Function]}
            >
              <div
                className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  disabled={false}
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  placeholder="Name your graph"
                  required={false}
                  type="text"
                  value="empty graph"
                />
                <fieldset
                  aria-hidden={true}
                  className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    className="css-1a1ihhj"
                  >
                    <span
                      className="notranslate"
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "&#8203;",
                        }
                      }
                    />
                  </legend>
                </fieldset>
              </div>
            </div>
            <br />
            <div
              className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
              onKeyPress={[Function]}
            >
              <div
                className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  disabled={false}
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  placeholder="Enter number to graph"
                  required={false}
                  type="number"
                  value=""
                />
                <fieldset
                  aria-hidden={true}
                  className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    className="css-1a1ihhj"
                  >
                    <span
                      className="notranslate"
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "&#8203;",
                        }
                      }
                    />
                  </legend>
                </fieldset>
              </div>
            </div>
            <br />
            <button
              className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-sghohy-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              Save
            </button>
          </form>
        </div>
        <div
          className="GraphChart"
        >
          <div>
            This is an EmptyGraph
          </div>
        </div>
      </div>
      <div
        className="right col-1-3"
      >
        <div
          className="GraphList"
        >
          <h3>
            Graph List
          </h3>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`shallow snapshot 1`] = `
<div
  className="Graphit"
>
  <div
    className="wrapper"
  >
    <div
      className="row"
    >
      <div
        className="left col-2-3"
      >
        <GraphInput
          addNewNumber={[MockFunction]}
          graphName="empty graph"
        />
        <GraphChart
          graphObject={
            GraphObject {
              "data": Array [],
              "name": "empty graph",
            }
          }
        />
      </div>
      <div
        className="right col-1-3"
      >
        <GraphList
          changeSelectedGraph={[MockFunction]}
          deleteGraph={[MockFunction]}
          graphNameList={Array []}
        />
      </div>
    </div>
  </div>
</div>
`;
